### PR TITLE
ADD: Feature - Internal Register + Verify by email

### PR DESCRIPTION
**Pull request by h2312**
Resolve : BE Internal Register && BE Verify account
Describe: Using nodemailer to send email, email contain a link to verify user. Using jwt to verify if user is valid.
Screen Shot 2020-12-15 at 08 52 01

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,11 @@
+JWT_SECRET=hiepnguyenv
+PORT=2312
+MONGO_DB=mongodb+srv://hiep:hiep2312@cluster0.dlqag.mongodb.net/assignment3db?retryWrites=true&w=majority
```
Comment (NHTNhi) [inline .env]: Change to port 5000

File: src/controllers/auth.controller.ts [hunk 2]
```diff
@@ -1,14 +1,47 @@
...
+
+  public verify = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    const token = req.params.token;
+    if (!token) {
+      next(new HttpException(400, 'We were unable to find a user for this token.'));
+    }
+    try {
+      const decoded = jwt.verify(token, process.env.TOKEN_SECRET) as DecodedToken;
+      console.log(decoded);
+      await this.authService.verify(decoded.id);
```
Comment (NHTNhi) [inline src/controllers/auth.controller.ts:37]: Remove console.log()

File: src/controllers/auth.controller.ts [hunk 3]
```diff
@@ -1,14 +1,47 @@
...
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public verify = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    const token = req.params.token;
+    if (!token) {
+      next(new HttpException(400, 'We were unable to find a user for this token.'));
+    }
```
Comment (NHTNhi) [inline src/controllers/auth.controller.ts]: Move to inside try{} catch{}

File: src/routes/auth.route.ts [hunk 4]
```diff
@@ -16,6 +17,8 @@ class AuthRoute implements Route {
   }
 
   private initializeRoutes() {
+    this.router.post(`${this.path}/register/internal`, validationMiddleware(CreateUserDto, 'body'), this.authController.register);
+    this.router.get(`/verify-account/:token`, this.authController.verify);
```
Comment (NHTNhi) [inline src/routes/auth.route.ts]: Change route to `${this.path}/auth/verify-account/:token`

**Review by NHTNhi — APPROVED** [review]

**Review by nguyennhuttoan — APPROVED** [review]
LGTM

**Pull request merged**